File: anomaly-tracking-api/AnomalyTracking.Model/Molds/Mold.cs
using AnomalyTracking.Model.AnomalyDeclarations;
using AnomalyTracking.Model.Cavities;
using Shared.Core.Model.Common;
using Shared.Core.Model.Core;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace AnomalyTracking.Model.Molds
{
    /// <summary>
    /// DTO that represents Mold object.
    /// </summary>
    [DataContract]
    public class Mold : EntityBase
    {
        /// <summary>
        /// Mold's label.
        /// </summary>
        [DataMember]
        public string Label { get; set; }
       


        /// <summary>
        /// Address
        /// </summary>
        [DataMember]
        public int ClientId { get; set; }


        /// <summary>
        /// Mold
        /// </summary>
        [DataMember]
        public IEnumerable<AnomalyDeclaration> AnomalyDeclarations { get; set; }

        /// <summary>
        /// Mold
        /// </summary>
        [DataMember]
        public IEnumerable<Cavity> Cavities { get; set; }
    }
}

File: anomaly-tracking-webapp/src/app/views/anomalytracking.domain/_shared/models/mold/mold.ts
import {
  EntityBase,
  IEntityBaseInfo,
} from 'src/app/views/core.domain/shared/models/common/entitybase';
import { Cavity } from '../cavity/cavity';

/***
 * Represents the mold object info.
 */
export interface IMoldInfo extends IEntityBaseInfo {
  Label: string;
  ClientId: number;
  Cavities: Cavity[];
}

/**
 * @Class mold
 * @classdesc Represents a mold object.
 */
export class Mold extends EntityBase {
  Label: string;
  ClientId: number;
  Cavities: Cavity[];

  /**
   * @constructor
   * Creates an instance of project.
   */
  constructor(info: IMoldInfo) {
    super(info);
    this.Label = info.Label;
    this.ClientId = info.ClientId;
    this.Cavities = info.Cavities;
  }
}
